wandb/sdk/artifacts/artifact.py

```diff
@@ -2475,29 +2475,24 @@ def link(
             # Wait until the artifact is committed before trying to link it.
             self.wait()
 
-        api = Api()
+        api = Api(overrides={"entity": self.source_entity})
```

Comment:
are there potentially other things that the user could have set when they initialized the API that could impact this code? (I'm not sure if this question really makes sense since I'm not an SDK expert)

Reply:
not sure if we should be setting this for the user..? why is it necessary to do this if we're using `self.source_entity` everywhere downstream? 

IIUC, the `_resolve_org_entity_name` favors the `entity` provided, and gathers the org entity name from it if `organization=""`? Or am i missing something?